### PR TITLE
use StreamResponse to download the log file

### LINKED ISSUE #14268: Separate lines in the downloadable owncloud.log
If you download the owncloud.log from `/index.php/settings/admin` the lines won't be separated causing the post output to be on one line instead of several different.

This should be fixed so that issue reports here on GitHub will be easier to read. 

CC @DeepDiver1975

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #14268

@MorrisJobke

### CODE REVIEW

**Comment by enoch85** [comment]
Applied the patch on 8.0.2, but got WSOD when I pressed the "download owncloud.log" button.

**Comment by enoch85** [comment]
```
Error	PHP	Class 'OCP\AppFramework\Http\StreamResponse' not found at /var/www/owncloud/settings/controller/logsettingscontroller.php#104	2015-03-27T08:10:59+00:00
```

**Comment by DeepDiver1975** [comment]
> Applied the patch on 8.0.2, but got WSOD when I pressed the "download owncloud.log" button.

this is a patch against master - OC8.1 - this will not work on OC8

**Comment by MorrisJobke** [comment]
Works :+1:

**Comment by MorrisJobke** [comment]
cc @LukasReschke Easy one ;)

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

File: tests/settings/controller/logsettingscontrollertest.php [hunk 1]
```diff
@@ -66,15 +66,9 @@ public function logLevelData() {
...
-	public function testGetFilenameForDownload() {
-		$timestamp = 42;
-		$this->container['Config']
-			->expects($this->once())
-			->method('getSystemValue')
-			->with('instanceid')
-			->will($this->returnValue('0xF'));
-		$filename = \Test_Helper::invokePrivate($this->logSettingsController, 'getFilenameForDownload', [$timestamp]);
+	public function testDownload() {
+		$response = \Test_Helper::invokePrivate($this->logSettingsController, 'download');
```
Comment (MorrisJobke) [inline tests/settings/controller/logsettingscontrollertest.php]: Magic doesn't needed. It's a public method ;)

Comment (DeepDiver1975) [inline tests/settings/controller/logsettingscontrollertest.php]: fixit

Comment (MorrisJobke) [inline tests/settings/controller/logsettingscontrollertest.php]: yes sir yes

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
The unit tests run locally.

**Comment by MorrisJobke** [comment]
Jenkins PR #15266

**Comment by icewind1991** [comment]
looks good :+1:

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Pull request merged**